File: python/abstractions-publish-gateway.py
#!/usr/bin/env python3
# abstractions-publish-gateway.js — portiert nach python
# Quelle: javascript, Projects@abstractions:javascript/abstractions-publish-gateway.js
# Erzeugt: 2026-08-08 durch ABSTRACTIONS_MANAGER.py

# abstractions-publish-gateway.sh — portiert nach javascript
# Quelle: shell, OpenClaw@gateway2:scripts/abstractions-publish-gateway.sh
# Erzeugt: 2026-08-07 durch ABSTRACTIONS_MANAGER.py

# Workspace-visible wrapper for the gateway publish job.
import subprocess
import sys
import os

script_path = os.path.join('/home/openclaw/.openclaw/scripts/abstractions-publish-gateway.sh')

try:
    result = subprocess.run([script_path] + sys.argv[1:])
    sys.exit(result.returncode)
except FileNotFoundError:
    print(f"Failed to start script: {script_path} not found", file=sys.stderr)
    sys.exit(1)
except Exception as err:
    print(f"Failed to start script: {err}", file=sys.stderr)
    sys.exit(1)

File: javascript/abstractions-publish-gateway.js
#!/usr/bin/env node
// abstractions-publish-gateway.pl — portiert nach javascript
// Quelle: perl5, Projects@abstractions:perl5/abstractions-publish-gateway.pl
// Erzeugt: 2026-08-08 durch ABSTRACTIONS_MANAGER.py

// abstractions-publish-gateway.sh — portiert nach perl5
// Quelle: shell, OpenClaw@gateway2:scripts/abstractions-publish-gateway.sh
// Erzeugt: 2026-08-07 durch ABSTRACTIONS_MANAGER.py

// Workspace-visible wrapper for the gateway publish job.

const { spawn } = require('child_process');
const path = require('path');

const scriptPath = path.join('/home/openclaw/.openclaw/scripts/abstractions-publish-gateway.sh');

// Forward all arguments to the shell script
const child = spawn(scriptPath, process.argv.slice(2), {
  stdio: 'inherit'
});

child.on('error', (err) => {
  console.error(`Fehler beim Ausführen des Skripts: ${err.message}`);
  process.exit(1);
});

child.on('close', (code) => {
  process.exit(code);
});
